API/Data/OperationRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using API.DTOs;
+using API.Entities;
+using API.Helpers;
+using API.Interfaces;
+using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API.Data
+{
+    public class OperationRepository : IOperationRepository
+    {
+        private readonly DataContext _context;
+        private readonly IMapper _mapper;
+        private readonly IBankAccountRepository _bankAccountRepository;
```

Comment:
Tak nie rób. Repozytoria to jedna warstwa - nie powinny miedzy soba rozmawiac. Przepyw sterowania to controller -> repo. Wtedy masz pewnosc ze nigdy nie zrobisz circurral reference. Teraz dodales bank account repo do operations. A jak bedziesz potrzebowal w banck accounts repo operations repo to nagle pojawia sie problemy bo nie bedzie sie dalo - oba obiekty beda potrzebowaly do stworzenia sie tego drugiego - deadlock